### PR TITLE
Update SmallRye Config to 2.3.0

### LINKED ISSUE #14847: Allow specifying optional config section with YAML configuration
**Describe the bug**
See [here](https://quarkusio.zulipchat.com/#narrow/stream/187030-sers/topic/Option.20configuration.20not.20working.3F)
**Expected behavior**
Optional configuration sections should be allowed with Quarkus YAML config.

**Actual behavior**
I know of no way of specifying optional section with YAML config.

**To Reproduce**
Unpack this project 
[issue_optional_config.zip](https://github.com/quarkusio/quarkus/files/5931906/issue_optional_config.zip)
and follow the instruction [here](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Option.20configuration.20not.20working.3F)

### LINKED ISSUE #15081: Validation on configuration using interfaces doesn't work
**Describe the bug**
According to https://quarkus.io/guides/config-reference#additional-notes-on-configproperties I should be able to validate nested configuration using interfaces but it doesn't seem to work.

**Expected behavior**
I would expect the application to fail when starting up

**Actual behavior**
The application starts up without error

**To Reproduce**
Reproducer application attached

[quarkus-config-sample.zip](https://github.com/quarkusio/quarkus/files/5982942/quarkus-config-sample.zip)

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #15081
- Fixes #14847

https://github.com/smallrye/smallrye-config/releases/tag/2.3.0

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gastaldi** [comment]
@radcortez can you rebase just so CI can be triggered again and we can certify that the failed test is not caused by this PR?

**Comment by radcortez** [comment]
> @radcortez can you rebase just so CI can be triggered again and we can certify that the failed test is not caused by this PR?

Sure.

File: extensions/arc/runtime/src/main/java/io/quarkus/arc/config/ConfigPrefix.java [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 /**
  * Uses in order to set the prefix for a configuration object at the injection point
+ *
+ * @deprecated Please, use {@link io.smallrye.config.ConfigMapping} instead. This will be removed in a future Quarkus
+ *             version.
  */
 @Qualifier
```
Comment (gsmet) [inline extensions/arc/runtime/src/main/java/io/quarkus/arc/config/ConfigPrefix.java:18]: Can you add `@Deprecated` here too?

**Review by gsmet — CHANGES_REQUESTED** [review]
@radcortez I spotted a small thing, could you adjust? Then we will merge.

File: extensions/arc/runtime/src/main/java/io/quarkus/arc/config/ConfigPrefix.java (hunk 1 above)
Comment (radcortez) [reply extensions/arc/runtime/src/main/java/io/quarkus/arc/config/ConfigPrefix.java:18]: Sure.

**Review by gsmet — APPROVED** [review]

**Pull request merged**